src/libs/OptionsListUtils.js

```diff
@@ -206,33 +212,53 @@ function hasReportDraftComment(report) {
 }
 
 /**
- * Get the Avatar urls or return the icon according to the chat type
+ * Returns the appropriate icons for the given chat report using the stored personalDetails.
+ * The Avatar sources can be URLs or Icon components according to the chat type.
  *
  * @param {Object} report
+ * @param {*} [defaultIcon]
  * @returns {Array<*>}
  */
-function getAvatarSources(report) {
-    return _.map(lodashGet(report, 'icons', ['']), (source) => {
-        if (source) {
-            return source;
-        }
-        if (ReportUtils.isArchivedRoom(report)) {
-            return Expensicons.DeletedRoomAvatar;
-        }
-        if (ReportUtils.isAdminRoom(report)) {
-            return Expensicons.AdminRoomAvatar;
-        }
-        if (ReportUtils.isAnnounceRoom(report)) {
-            return Expensicons.AnnounceRoomAvatar;
-        }
-        if (ReportUtils.isChatRoom(report)) {
-            return Expensicons.ActiveRoomAvatar;
-        }
-        if (ReportUtils.isPolicyExpenseChat(report)) {
-            return Expensicons.Workspace;
+function getReportIcons(report, defaultIcon = null) {
```

Comment:
Unsure how hard this would be to change, but this feels like more of a `ReportUtils` responsibility maybe?

Looking at what `OptionsListUtils` is doing and it's unclear whether this method belongs here 🤔

Reply:
Fair point, yeah I leave it there because it was always in that file, but makes more sense to move it to `ReportUtils`